release: Add release notes for Metals 0.11.8 (#4214)
scalameta/metals: website/blog/2022-08-08-aluminium.md

---
author: Rikito Taniguchi
title: Metals v0.11.8 - Aluminium
authorURL: https://twitter.com/tanishiking
authorImageURL: https://github.com/tanishiking.png
---

We're happy to announce the release of Metals v0.11.8, bringing a number of improvements for both Scala 2 and Scala 3.

<table>
<tbody>
  <tr>
    <td>Commits since last release</td>
    <td align="center">81</td>
  </tr>
  <tr>
    <td>Merged PRs</td>
    <td align="center">77</td>
  </tr>
    <tr>
    <td>Contributors</td>
    <td align="center">15</td>
  </tr>
  <tr>
    <td>Closed issues</td>
    <td align="center">22</td>
  </tr>
  <tr>
    <td>New features</td>

>     <td align="center">6</td>

Reviewer: Let's make sure the numbers are valid

```suggestion
    <td align="center">5</td>
```

  </tr>
</tbody>
</table>

For full details: https://github.com/scalameta/metals/milestone/52?closed=1

Metals is a language server for Scala that works with VS Code, Vim, Emacs and
Sublime Text. Metals is developed at the
[Scala Center](https://scala.epfl.ch/) and [VirtusLab](https://virtuslab.com)
with the help from [Lunatech](https://lunatech.com) along with contributors from
the community.

## TL;DR

Check out [https://scalameta.org/metals/](https://scalameta.org/metals/), and
give Metals a try!

- [Scala 3] Auto import and completion for extension methods
- [Scala 3] Convert to Named Parameters code action
- [Scala 3] Scaladoc Completion for Scala3
- [Scala 3] Completions in string interpolation
- [Scala 2] Automatic import of types in string interpolations

> - More stable code navigation across subprojects

Reviewer: Should we remove it since we no longer explain it? Or have it under an additional section for bugfixes?

## [Scala 3] Scaladoc completion

Metals now supports offering Scaladoc completions in Scala 3. When typing `/**` you get an option to auto-complete a scaladoc template for methods, classes, etc.!

![scala-doc-completion](https://i.imgur.com/MEJUXr3.gif)

## [Scala 3] Completions in string interpolation

In the previous versions, whenever users wanted to include a value in a string using string interpolation, they would need to do it all manually. Now, it is possible to get an automatic conversion to string interpolation when typing `$value`, as well as automatic wrapping in `{}` when accessing members of such value.

![scala3-interpolation](https://i.imgur.com/EyFKpiv.gif)

## [Scala 2] Automatically import types in string interpolations

Previously, the only suggestions for string interpolations were coming from the currently available symbols in scope. This meant that if you wanted to import something from another package, you would need to do it manually.

This problem is now resolved. Users can easily get such symbols automatically imported, which creates a seamless workflow.

![scala2-inteprolation](https://i.imgur.com/cCWTQnj.gif)

> The feature also works for Scala 3.

Reviewer: ```suggestion
The feature also being worked on for Scala 3.
```
I haven't actually finished it :sweat:

Author: fixed 😄 https://github.com/scalameta/metals/pull/4214/commits/9620b840f2de62f10028678d1db7f6963e90f609

## Code Action documentation

Have you ever wondered what kind of refactorings are available in Metals?
Check out this new page in the documentation! You can see a list of all the code actions in Metals with examples.
https://scalameta.org/metals/docs/codeactions/codeactions

Big thanks to [zmerr](https://github.com/vzmerr) for writing this documentation.

## Contributors

Big thanks to everybody who contributed to this release or reported an issue!

```
$ git shortlog -sn --no-merges v0.11.7..v0.11.8
31	Tomasz Godzik
    10	Rikito Taniguchi
     9	Scala Steward
     6	jkciesluk
     6	Kamil Podsiadło
     5	vzmerr
     3	Vadim Chelyshov
     2	Adrien Piquerez
     2	scalameta-bot
     2	zmerr
     1	Arthur S
     1	Anton Sviridov
     1	tgodzik
     1	Scalameta Bot
     1	Chris Kipp
```

## Merged PRs

## [v0.11.8](https://github.com/scalameta/metals/tree/v0.11.8) (2022-08-08)

[Full Changelog](https://github.com/scalameta/metals/compare/v0.11.7...v0.11.8)

**Merged pull requests:**

> - chore(ci): small changes to account for migration from LSIF -> SCIP

Reviewer: Let's update the list before the release.

---
author: Rikito Taniguchi
title: Metals v0.11.8 - Aluminium
authorURL: https://twitter.com/tanishiking
authorImageURL: https://github.com/tanishiking.png
---

> We're happy to announce the release of Metals v0.11.8, which brings a number of improvements for both Scala 2 and Scala 3.

Reviewer 2: ```suggestion
We're happy to announce the release of Metals v0.11.8, bringing a number of improvements for both Scala 2 and Scala 3.
```
nit

<table>
<tbody>
  <tr>
    <td>Commits since last release</td>
    <td align="center">81</td>
  </tr>
  <tr>
    <td>Merged PRs</td>
    <td align="center">77</td>
  </tr>
    <tr>
    <td>Contributors</td>
    <td align="center">15</td>
  </tr>
  <tr>
    <td>Closed issues</td>
    <td align="center">22</td>
  </tr>
  <tr>
    <td>New features</td>
    <td align="center">6</td>
  </tr>
</tbody>
</table>

For full details: https://github.com/scalameta/metals/milestone/52?closed=1

Metals is a language server for Scala that works with VS Code, Vim, Emacs and
Sublime Text. Metals is developed at the
[Scala Center](https://scala.epfl.ch/) and [VirtusLab](https://virtuslab.com)
with the help from [Lunatech](https://lunatech.com) along with contributors from
the community.

## TL;DR

Check out [https://scalameta.org/metals/](https://scalameta.org/metals/), and
give Metals a try!

- [Scala 3] Type completion
- [Scala 3] Auto import and completion for extension methods
- [Scala 3] Convert to Named Parameters code action
- [Scala 3] Scaladoc Completion for Scala3
- [Scala 3] Completions in string interpolation

> - [Scala 2] Automatically import types in string interpolations

Reviewer 2: ```suggestion
- [Scala 2] Automatic import of types in string interpolations
```
because all the previous items are phrases, rather than sentences

NIT

- More stable code navigation across subprojects
- Code Action documentation

> - Support Scala 3.2.0-RC3, Scala 3.2.0-RC2

Reviewer 2: ```suggestion
- Support of Scala 3.2.0-RC3, Scala 3.2.0-RC2
```

Reviewer 2: to make it a phrase

and a lot of bugfixes!

## [Scala 3] Type Competion

![type-completion](https://i.imgur.com/7kDzoLs.gif)

> This newly implemented feature for Scala 3 is not only adding necessary `[]` and `{}` suffixes to type completions; but also, it is filtering out some of the invalid completion results.

Reviewer 3: ```suggestion
This newly implemented feature for Scala 3 is not only adding necessary `[]` and `{}` suffixes to type completions; but also filtering out some of the invalid completion results.
```

> As this is a new feature, please let us know if you encounter any missing or invalid results, by opening an issue.

Reviewer 3: ```suggestion
As this is a new feature, please let us know if you encounter any missing or invalid results by opening an issue.
```

> As this is a new feature so please let us know if you encounter any missing or invalid results, by opening an issue.

Reviewer 2: ```suggestion
As this is a new feature, please let us know if you encounter any missing or invalid results, by opening an issue.
```
to make it grammatically correct. without duplicate sentence connector.

## [Scala 3] Auto import and completion for extension methods

> Scala3 introduced `extension methods` that allow defining new methods to your existing types.

Reviewer 2: ```suggestion
You might know that Scala 3 has introduced `extension methods` that allow defining new methods to your existing types.
```
To make it clear that this is just an introductory sentence, rather than the main message.

Author: Sounds better! 👍

> However, previously, Metals couldn't auto-complete extension methods, and developers have to find an appropriate extension method from their workspace and import it by hand. It was time-consuming and not always beginner friendly.

Reviewer 2: ```suggestion
However, previously, Metals couldn't auto-complete extension methods; so developers had to find an appropriate extension method from their workspace and manually import it. But, this was time-consuming and not always beginner friendly.
```

> However, previously, Metals couldn't auto-complete extension methods; so developers had to find an appropriate extension method from their workspace and manually import it. But, this was time-consuming and not always beginner friendly.

Reviewer 3: ```suggestion
Previously, Metals couldn't auto-complete extension methods; so developers had to find an appropriate extension method from their workspace and manually import it. But, this was time-consuming and not always beginner friendly.
```

Now, Metals provides auto-completion for extension methods and automatically imports them!

![extension-methods](https://i.imgur.com/EAbVHeH.gif)

## [Scala 3] Convert to Named Parameters code action

[Metals 0.11.7 added `ConvertToNamedParameters` code action to Scala2](https://scalameta.org/metals/blog/2022/07/04/aluminium#scala-2-add-converttonamedarguments-code-action).

Thanks to the contribution by [@jkciesluk](https://github.com/jkciesluk), this feature is now available for Scala 3!

![convert-to-named](https://i.imgur.com/9i7MWoQ.gif)

## [Scala 3] Scaladoc completion

> Metals now supports the offering of Scaladoc completions in Scala3. When typing `/**` you get an option to auto-complete a scaladoc template for methods, classes, etc.!

Reviewer 2: ```suggestion
Metals now supports the offering of Scaladoc completions in Scala 3. When typing `/**` you get an option to auto-complete a scaladoc template for methods, classes, etc.!
```

> Metals now supports the offering of Scaladoc completions in Scala 3. When typing `/**` you get an option to auto-complete a scaladoc template for methods, classes, etc.!

Reviewer 3: ```suggestion
Metals now supports offering Scaladoc completions in Scala 3. When typing `/**` you get an option to auto-complete a scaladoc template for methods, classes, etc.!
```

![scala-doc-completion](https://i.imgur.com/MEJUXr3.gif)

## [Scala 3] Completions in string interpolation

> In the previous versions, whenever users wanted to include a value in a string using string interpolation, they would have needed to do it all manually. Now it is possible to get an automatic conversion to string interpolation when users type `$value` as well as automatic wrapping in `{}` when accessing members of such value.

Reviewer 2: ```suggestion
In the previous versions, whenever users wanted to include a value in a string using string interpolation, they would have needed to do it all manually. Now, it is possible to get an automatic conversion to string interpolation when typing `$value`, as well as automatic wrapping in `{}` when accessing members of such value.
```

> In the previous versions, whenever users wanted to include a value in a string using string interpolation, they would have needed to do it all manually. Now, it is possible to get an automatic conversion to string interpolation when typing `$value`, as well as automatic wrapping in `{}` when accessing members of such value.

Reviewer 3: ```suggestion
In the previous versions, whenever users wanted to include a value in a string using string interpolation, they would need to do it all manually. Now, it is possible to get an automatic conversion to string interpolation when typing `$value`, as well as automatic wrapping in `{}` when accessing members of such value.
```

![scala3-interpolation](https://i.imgur.com/EyFKpiv.gif)

## [Scala 2] Automatically import types in string interpolations

> Previously, the only suggestions for string interpolations was coming from the currently available symbols in scope. This meant that if you wanted to bring something from another package, you would have needed to do it manually.

Reviewer 2: ```suggestion
Previously, the only suggestions for string interpolations were coming from the currently available symbols in scope. This meant that if you wanted to import something from another package, you would have needed to do it manually.
```

> Previously, the only suggestions for string interpolations were coming from the currently available symbols in scope. This meant that if you wanted to import something from another package, you would have needed to do it manually.

Reviewer 3: ```suggestion
Previously, the only suggestions for string interpolations were coming from the currently available symbols in scope. This meant that if you wanted to import something from another package, you would need to do it manually.
```

> This problem is now resolved. Users can easily get symbols automatically imported as if working in a normal code.

Reviewer 2: ```suggestion
This problem is now resolved. Users can easily get such symbols automatically imported, which creates a seamless workflow.
```
